src/scripts/extensions/authenticationHelper.ts

```diff
@@ -139,4 +143,43 @@ export class AuthenticationHelper {
 		// Note that we are returning true by default to ensure the N-1 scenario.
 		return userInfo.cookieInRequest !== undefined ? userInfo.cookieInRequest : true;
 	}
+
+	/**
+	 * fetch the user data bounday from the emailAddress
+	 * @param userInfo 
+	 * @returns user data boudary
+	 */
+	private async getUserDataBoundary(userInfo: UserInfoData): Promise<string | undefined> {
+		try {
+			if (!userInfo) {
+				return undefined;
+			}
+			if (userInfo.authType === AuthType[AuthType.Msa]) {
+				return DataBoundary[DataBoundary.GLOBAL];
+			}
+			let domainValue;
+			if (!userInfo.emailAddress) {
+				return undefined;
+			} else {
+				domainValue = userInfo.emailAddress.substring(
+					userInfo.emailAddress.indexOf("@") + 1
+				);
+			}
+			const urlDataBoundaryDomain: string = UrlUtils.addUrlQueryValue(Constants.Urls.userDataBoundaryDomain, Constants.Urls.QueryParams.domain, domainValue);
+			let response = await fetch(urlDataBoundaryDomain, {
```

Comment:
What would this API return in case of bad domain  value? Will it route to EU regions for it to be most compliant?

Reply:
In Case of bad domain value there wont be any telemetry region field in the response to this API. So used UNKNOWN as telemetry region  and this would confirm to EU region as per service changes